spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/JobCommands.java

```diff
@@ -0,0 +1,277 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.shell.command;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.rest.client.JobOperations;
+import org.springframework.cloud.dataflow.rest.resource.JobExecutionResource;
+import org.springframework.cloud.dataflow.rest.resource.JobInstanceResource;
+import org.springframework.cloud.dataflow.rest.resource.StepExecutionProgressInfoResource;
+import org.springframework.cloud.dataflow.rest.resource.StepExecutionResource;
+import org.springframework.cloud.dataflow.shell.config.DataFlowShell;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.shell.table.Table;
+import org.springframework.shell.table.TableBuilder;
+import org.springframework.shell.table.TableModelBuilder;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+/**
+ * Job commands for the SCDF Shell.
+ *
+ * @author Glenn Renfro
+ */
+@Component
+public class JobCommands implements CommandMarker {
+
+	private static final String EXECUTION_VIEW = "job execution view";
+
+	private static final String EXECUTION_LIST = "job execution list";
+
+	private static final String STEP_EXECUTION_LIST = "job execution step list";
+
+	private static final String INSTANCE_VIEW = "job instance view";
+
+	private static final String STEP_EXECUTION_PROGRESS = "job execution step progress";
+
+	private static final String STEP_EXECUTION_VIEW = "job execution step view";
+
+	@Autowired
+	private DataFlowShell dataFlowShell;
+
+	@CliCommand(value = EXECUTION_LIST,
+			help = "List created job executions filtered by jobName")
+	public Table executionList(
+			@CliOption(key = { "name" },
+					help = "the job name to be used as a filter",
+					mandatory = false) String name) {
+
+		final PagedResources<JobExecutionResource> jobs;
+		if (name == null) {
+			jobs = jobOperations().executionList();
+		}
+		else {
+			jobs = jobOperations().executionListByJobName(name);
+		}
+
+		TableModelBuilder<Object> modelBuilder = new TableModelBuilder<>();
```

Comment:
Did consider it, however in the cases where it could be used, there were tweaks to the data being written to the table that the BeanListTableModel did not support.